service/frontend/workflowHandler.go

```diff
@@ -3706,6 +3717,69 @@ func isFailoverRequest(updateRequest *gen.UpdateDomainRequest) bool {
 	return updateRequest.ReplicationConfiguration != nil && updateRequest.ReplicationConfiguration.ActiveClusterName != nil
 }
 
+func isGraceFailoverRequest(updateRequest *gen.UpdateDomainRequest) bool {
+	return updateRequest.IsSetFailoverTimeoutInSeconds()
+}
+
+func (wh *WorkflowHandler) checkOngoingFailover(
+	ctx context.Context,
+	domainName *string,
+) error {
+
+	clusterMetadata := wh.GetClusterMetadata()
+	respChan := make(chan *gen.DescribeDomainResponse)
+	wg := &sync.WaitGroup{}
+
+	decribeDomain := func(
+		ctx context.Context,
+		client frontend.Client,
+		domainName *string,
+	) {
+		defer wg.Done()
+		resp, _ := client.DescribeDomain(
+			ctx,
+			&gen.DescribeDomainRequest{
+				Name: domainName,
+			},
+		)
+		respChan <- resp
+	}
+
+	for clusterName, cluster := range clusterMetadata.GetAllClusterInfo() {
+		if !cluster.Enabled {
+			continue
+		}
+		frontendClient := wh.GetRemoteFrontendClient(clusterName)
+		wg.Add(1)
+		go decribeDomain(
+			ctx,
+			frontendClient,
+			domainName,
+		)
+	}
+	wg.Wait()
+	close(respChan)
+
+	var failoverVersion *int64
+	for resp := range respChan {
+		if resp == nil {
+			return &gen.InternalServiceError{
+				Message: "Failed to verify failover version from all clusters",
+			}
+		}
+		if failoverVersion == nil {
+			failoverVersion = resp.FailoverVersion
+		}
+		if failoverVersion != resp.FailoverVersion {
+			prevCluster := clusterMetadata.ClusterNameForFailoverVersion(*resp.FailoverVersion)
```

Comment:
I don't quite understand this. What does `previous` mean here?

Reply:
This only used for log message. I remove this part.